lib/rubocop/cop/lint/empty_when.rb

```diff
@@ -23,16 +23,38 @@ def on_case(node)
           end
         end
 
+        def comment_lines
+          @comment_lines ||= processed_source.comments.map{|comment| comment.loc.line}
+        end
+
         private
 
         def check_when(when_node)
           return unless empty_when_body?(when_node)
-
           add_offense(when_node, when_node.source_range, MSG)
         end
 
         def empty_when_body?(when_node)
-          !when_node.to_a.last
+          !when_node.to_a.last && no_comments?(when_node)
+        end
+
+        def no_comments?(when_node)
+          (comment_lines & line_range_to_next_node(when_node)).empty?
+        end
+
+        def line_range_to_next_node(node)
```

Comment:
Sorry, I don't know such an API.
I expected that I could find it from comment_config, but it did not seem to exist. https://github.com/bbatsov/rubocop/blob/456c6d784d097981ceedaf98d7ad0e396e603160/lib/rubocop/comment_config.rb